easybuild/framework/easyconfig/types.py

```diff
@@ -375,6 +375,43 @@ def to_list_of_strings_and_tuples_and_dicts(spec):
     return str_tup_list
 
 
+def to_sanity_check_paths_entry(spec):
+    """
+    Convert a 'list of lists and strings' to a 'list of tuples and strings' while allowing dicts of lists or strings
+
+    Example:
+        ['foo', ['bar', 'baz'], {'f42': ['a', 'b']}]
+        to
+        ['foo', ('bar', 'baz'), {'f42': ('a', 'b')}]
+    """
+    result = []
+
+    if not isinstance(spec, (list, tuple)):
+        raise EasyBuildError("Expected value to be a list, found %s (%s)", spec, type(spec))
+
+    for elem in spec:
+        if isinstance(elem, (string_type, tuple)):
+            result.append(elem)
+        elif isinstance(elem, list):
+            result.append(tuple(elem))
+        elif isinstance(elem, dict):
+            result.append({})
+            for key, value in elem.items():
+                if not isinstance(key, string_type):
+                    raise EasyBuildError("Expected keys to be of type string, got %s (%s)", key, type(key))
+                if isinstance(value, (string_type, tuple)):
+                    result[-1][key] = value
+                elif isinstance(value, list):
+                    result[-1][key] = tuple(value)
+                else:
+                    raise EasyBuildError("Expected elements to be of type string, tuple or list, got %s (%s)",
+                                         value, type(value))
```

Comment:
What about this even easier solution:
```suggestion
            for key, value in elem.items():
                if not isinstance(key, string_type):
                    raise EasyBuildError("Expected keys to be of type string, got %s (%s)", key, type(key))
                elif isinstance(value, list):
                    elem[key] = tuple(value)
                elif not isinstance(value, (string_type, tuple)):
                    raise EasyBuildError("Expected elements to be of type string, tuple or list, got %s (%s)",
                                         value, type(value))
                result.append(elem)
```

Reply:
Yeah that's better :-)